pysindy/optimizers/trapping_sr3.py

```diff
@@ -310,14 +441,47 @@ def _reduce(self, x, y):
         """
 
         n_samples, n_features = x.shape
-        r = (self.PL).shape[0]
-        N = n_features
+        r = y.shape[1]
+        N = int((r ** 2 + 3 * r) / 2.0)
+
+        # If these tensors are not passed, or incorrect shape, assume zeros
+        if self.PQ is None:
+            self.PQ = np.zeros((r, r, r, r, n_features))
+        elif (self.PQ).shape != (r, r, r, r, n_features) and (self.PQ).shape != (
+            r,
+            r,
+            r,
+            r,
+            N,
+        ):
+            self.PQ = np.zeros((r, r, r, r, n_features))
+        if self.PL is None:
+            self.PL = np.zeros((r, r, r, n_features))
+        elif (self.PL).shape != (r, r, r, n_features) and (self.PL).shape != (
+            r,
+            r,
+            r,
+            N,
+        ):
+            self.PL = np.zeros((r, r, r, n_features))
+
+        # Check if the tensor symmetries are properly defined
+        if self._bad_PL(self.PL):
+            raise ValueError("PL tensor was passed but not properly")
+        if self._bad_PQ(self.PQ):
+            raise ValueError("PQ tensor was passed but not properly")
+
+        # If PL/PQ finite and correct, so trapping theorem is being used,
+        # then make sure library is quadratic and correct shape
+        print(np.any(self.PL != 0.0), np.any(self.PQ != 0.0), n_features, r, N)
```

Comment:
Is this line needed? If it is, could you make it more informative by also printing a description of what's being printed?

Reply:
the print statement was for debugging, it can be deleted. The check for a finite PL/PQ is elsewhere

Reply:
Cool, I can delete it along with some other minor local changes I've made.